### DISCUSSION TITLE
surface temperature (TgOut), 3D wall temperature

### DISCUSSION THREAD

**Discussion by kg85112** [UMEP-dev/UMEP#630] [Q&A]
Hello,

My name is G. Kang, and I am conducting research on urban greening modeling using CFD models.

Following up on my previous question (#593, regarding building wall temperatures), I have made some modifications to the SOLWEIG code to calculate 3D surface temperatures.

I have a few questions:

1. When extracting TgOut, the corrected (CI_TgG, shadow) ground/roof temperature values are displayed, but the values stored do not account for the surface properties (land cover) and GVF. Is this intentional? 

I recalculated the TgOut values using the Lup values, which were computed considering land cover and GVF. Please see the illustration below:

* original: 
TgTemp = Tg * shadow + Ta
TgOut, timeadd, TgOut1 = TsWaveDelay_2015a(TgTemp, firstdaytime, timeadd, timestepdec, TgOut1)

* renew:
renew_TgTemp = ((Lup+LupE+LupS+LupW+LupN)/(5*SBC*emis_grid))**(0.25) - 273.15

![GK](https://github.com/UMEP-dev/UMEP/assets/164312008/d1f48f05-0053-4105-a8dc-b21898ec715f)

2. Regarding 3D Wall temperature:
I calculated the 3D wall temp., referring to the SEBE_2015a_calc_forprocessing.py code and TgTemp and TgOut variables. I considered various factors such as walls, wallsun, wallsh, and wallshve.

Could you please let me know if there are any additional considerations or precautions for calculating 3D wall temperatures in the 2024a version mentioned earlier?

Thank you so much all the time! 

![Tsurf_1](https://github.com/UMEP-dev/UMEP/assets/164312008/c316137f-4834-4908-a605-6f1561f7523c)
![Tsurf_2](https://github.com/UMEP-dev/UMEP/assets/164312008/efcd3582-6f04-4acf-aeff-e8b344f8e885)

**Comment by nilswallenberg** [comment]
Hi,

Not sure I understand. Can you please elaborate? Regarding the walls we are currently working on an entirely new scheme that includes voxels, i.e. we will not use Tg or TgOut. We will however include walls, wallsun and wallsh.

Cheers,
Nils

**Comment by kg85112** [comment]
Dear Nils,

Thank you for taking the time.

I have a question regarding the use of POIs and the saved surface temperature (TgOut). 
It appears that TgOut values are the same across different areas.

 As shown in the first figure, the surface temperature of sunlit areas on both building rooftops and the ground is identical, and the surface temperature of shaded areas is also the same.

Upon reviewing the code (Solweig_2022a_calc_forprocessing.py), it seems that land cover is not considered when calculating TgOut. Instead, land cover is used in the subsequent step when calculating long wave radiation.

Since I require surface temperature values that account for land cover variations, I would like to ask for your guidance on this matter.

* In Solweig_2022a_calc_forprocessing.py:

        Tg = Tg * CI_TgG  # new estimation
        Tgwall = Tgwall * CI_TgG
        if landcover == 1:
            Tg[Tg < 0] = 0  # temporary for removing low Tg during morning 20130205

        # # # # Ground View Factors # # # #
        gvfLup, gvfalb, gvfalbnosh, gvfLupE, gvfalbE, gvfalbnoshE, gvfLupS, gvfalbS, gvfalbnoshS, gvfLupW, gvfalbW,\
        gvfalbnoshW, gvfLupN, gvfalbN, gvfalbnoshN, gvfSum, gvfNorm = gvf_2018a(wallsun, walls, buildings, scale, shadow, first,
                second, dirwalls, Tg, Tgwall, Ta, emis_grid, ewall, alb_grid, SBC, albedo_b, rows, cols,
                                                                 Twater, lc_grid, landcover)

        # # # # Lup, daytime # # # #
        # Surface temperature wave delay - new as from 2014a
        Lup, timeaddnotused, Tgmap1 = TsWaveDelay_2015a(gvfLup, firstdaytime, timeadd, timestepdec, Tgmap1)
        LupE, timeaddnotused, Tgmap1E = TsWaveDelay_2015a(gvfLupE, firstdaytime, timeadd, timestepdec, Tgmap1E)
        LupS, timeaddnotused, Tgmap1S = TsWaveDelay_2015a(gvfLupS, firstdaytime, timeadd, timestepdec, Tgmap1S)
        LupW, timeaddnotused, Tgmap1W = TsWaveDelay_2015a(gvfLupW, firstdaytime, timeadd, timestepdec, Tgmap1W)
        LupN, timeaddnotused, Tgmap1N = TsWaveDelay_2015a(gvfLupN, firstdaytime, timeadd, timestepdec, Tgmap1N)
        
        # # For Tg output in POIs
        TgTemp = Tg * shadow + Ta
        TgOut, timeadd, TgOut1 = TsWaveDelay_2015a(TgTemp, firstdaytime, timeadd, timestepdec, TgOut1) #timeadd only here v2021a


Thank you for your assistance.
Best regards

**Reply by biglimp** [reply]
Yes, the land cover should be considered as Tg is a spatial variable create [here ](https://github.com/UMEP-dev/UMEP-processing/blob/60d6f35459031a6b3ed655f716bcf829c8076df2/processor/solweig_algorithm.py#L901).

Back calculation from Lup is not the same as Lup is considered from the center of gravitation of a person 1.1 meter above ground and Tg is on the ground surface. In Lup the ground view factor component is considered (see Lindberg and Grimmond 2011, DOI 10.1007/s00704-010-0382-8). When I look at your figure 1 I see that different ground cover produce different Tg based on land cover.

**Reply by kg85112** [reply]
Ah! Thank you very much.

I realize now that I was mistaken. I had assumed that the 'black asphalt surface temperature' should be higher than the 'building roof temperature', so I thought that while the land cover was differentiated, it was not reflected in Tg.

However, after following your advice and checking the code, I found that the values for Roofs(buildings) and Dark_asphalt were the same, which is why the surface temperatures were not differentiated.

Thank you so much!

Best regards,
biglimp.
